text/0000-default-and-expanded-rustc-errors.md

```diff
@@ -0,0 +1,290 @@
+- Feature Name: default_and_expanded_errors_for_rustc
+- Start Date: 2016-06-07
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+This RFC proposes an update to error reporting in rustc.  Its focus is to change the format of Rust error messages and improve --explain capabilities to focus on the user's code.  The end goal is for errors and explain text to be more readable, more friendly to new users, while still helping Rust coders fix bugs as quickly as possible.  We expect to follow this RFC with a supplemental RFC that provides a writing style guide for error messages and explain text with a focus on readability and education.
```

Comment:
nit: could you wrap at 80 chars please (I'm sure this was specified somewhere, but maybe it is just a de facto thing that people do?)